diff -ruN openssl-1.0.0b/Configure openssl-1.0.0b-patched/Configure
--- openssl-1.0.0b/Configure	2010-11-11 17:23:19 +0200
+++ openssl-1.0.0b-patched/Configure	2010-11-17 23:19:46 +0200
@@ -491,12 +491,12 @@
 #
 # Win64 targets, WIN64I denotes IA-64 and WIN64A - AMD64
 "VC-WIN64I","cl:-W3 -Gs0 -Gy -nologo -DOPENSSL_SYSNAME_WIN32 -DWIN32_LEAN_AND_MEAN -DL_ENDIAN -DUNICODE -D_UNICODE -D_CRT_SECURE_NO_DEPRECATE:::WIN64I::SIXTY_FOUR_BIT RC4_CHUNK_LL DES_INT EXPORT_VAR_AS_FN:ia64cpuid.o:ia64.o::aes_core.o aes_cbc.o aes-ia64.o::md5-ia64.o:sha1-ia64.o sha256-ia64.o sha512-ia64.o:::::::ias:win32",
-"VC-WIN64A","cl:-W3 -Gs0 -Gy -nologo -DOPENSSL_SYSNAME_WIN32 -DWIN32_LEAN_AND_MEAN -DL_ENDIAN -DUNICODE -D_UNICODE -D_CRT_SECURE_NO_DEPRECATE:::WIN64A::SIXTY_FOUR_BIT RC4_CHUNK_LL DES_INT EXPORT_VAR_AS_FN:x86_64cpuid.o:bn_asm.o x86_64-mont.o::aes-x86_64.o::md5-x86_64.o:sha1-x86_64.o sha256-x86_64.o sha512-x86_64.o::rc4-x86_64.o:::wp-x86_64.o:cmll-x86_64.o cmll_misc.o:auto:win32",
+"VC-WIN64A","cl:-Zi -W3 -Gs0 -Gy -nologo -DOPENSSL_SYSNAME_WIN32 -DWIN32_LEAN_AND_MEAN -DL_ENDIAN -DUNICODE -D_UNICODE -D_CRT_SECURE_NO_DEPRECATE:::WIN64A::SIXTY_FOUR_BIT RC4_CHUNK_LL DES_INT EXPORT_VAR_AS_FN:x86_64cpuid.o:bn_asm.o x86_64-mont.o::aes-x86_64.o::md5-x86_64.o:sha1-x86_64.o sha256-x86_64.o sha512-x86_64.o::rc4-x86_64.o:::wp-x86_64.o:cmll-x86_64.o cmll_misc.o:auto:win32",
 "debug-VC-WIN64I","cl:-W3 -Gs0 -Gy -Zi -nologo -DOPENSSL_SYSNAME_WIN32 -DWIN32_LEAN_AND_MEAN -DL_ENDIAN -DUNICODE -D_UNICODE -D_CRT_SECURE_NO_DEPRECATE:::WIN64I::SIXTY_FOUR_BIT RC4_CHUNK_LL DES_INT EXPORT_VAR_AS_FN:ia64cpuid.o:ia64.o::aes_core.o aes_cbc.o aes-ia64.o::md5-ia64.o:sha1-ia64.o sha256-ia64.o sha512-ia64.o:::::::ias:win32",
 "debug-VC-WIN64A","cl:-W3 -Gs0 -Gy -Zi -nologo -DOPENSSL_SYSNAME_WIN32 -DWIN32_LEAN_AND_MEAN -DL_ENDIAN -DUNICODE -D_UNICODE -D_CRT_SECURE_NO_DEPRECATE:::WIN64A::SIXTY_FOUR_BIT RC4_CHUNK_LL DES_INT EXPORT_VAR_AS_FN:x86_64cpuid.o:bn_asm.o x86_64-mont.o::aes-x86_64.o::md5-x86_64.o:sha1-x86_64.o sha256-x86_64.o sha512-x86_64.o::rc4-x86_64.o:::wp-x86_64.o:cmll-x86_64.o cmll_misc.o:auto:win32",
 # x86 Win32 target defaults to ANSI API, if you want UNICODE, complement
 # 'perl Configure VC-WIN32' with '-DUNICODE -D_UNICODE'
-"VC-WIN32","cl:-W3 -WX -Gs0 -GF -Gy -nologo -DOPENSSL_SYSNAME_WIN32 -DWIN32_LEAN_AND_MEAN -DL_ENDIAN -D_CRT_SECURE_NO_DEPRECATE:::WIN32::BN_LLONG RC4_INDEX EXPORT_VAR_AS_FN ${x86_gcc_opts}:${x86_asm}:win32n:win32",
+"VC-WIN32","cl:-Zi -W3 -WX -Gs0 -GF -Gy -nologo -DOPENSSL_SYSNAME_WIN32 -DWIN32_LEAN_AND_MEAN -DL_ENDIAN -D_CRT_SECURE_NO_DEPRECATE:::WIN32::BN_LLONG RC4_INDEX EXPORT_VAR_AS_FN ${x86_gcc_opts}:${x86_asm}:win32n:win32",
 # Unified CE target
 "debug-VC-WIN32","cl:-W3 -WX -Gs0 -GF -Gy -Zi -nologo -DOPENSSL_SYSNAME_WIN32 -DWIN32_LEAN_AND_MEAN -DL_ENDIAN -D_CRT_SECURE_NO_DEPRECATE:::WIN32::BN_LLONG RC4_INDEX EXPORT_VAR_AS_FN ${x86_gcc_opts}:${x86_asm}:win32n:win32",
 "VC-CE","cl::::WINCE::BN_LLONG RC4_INDEX EXPORT_VAR_AS_FN ${x86_gcc_opts}:${no_asm}:win32",
diff -ruN openssl-1.0.0b/build.bat openssl-1.0.0b-patched/build.bat
--- openssl-1.0.0b/build.bat	1970-01-01 02:00:00 +0200
+++ openssl-1.0.0b-patched/build.bat	2010-11-17 22:56:54 +0200
@@ -0,0 +1,26 @@
+call "%VCINSTALLDIR%\vcvarsall.bat" x86
+perl Configure VC-WIN32
+call ms\do_nasm debug
+nmake -f ms\nt.mak
+@if errorlevel 1 goto end
+call ms\do_nasm
+nmake -f ms\nt.mak
+@if errorlevel 1 goto end
+call "%VCINSTALLDIR%\vcvarsall.bat" x86_amd64
+perl Configure VC-WIN64A
+call ms\do_win64a debug
+nmake -f ms\nt.mak
+@if errorlevel 1 goto end
+call ms\do_win64a
+nmake -f ms\nt.mak
+@if errorlevel 1 goto end
+xcopy /i /y out32\*.lib ..\lib
+xcopy /i /y out32\*.pdb ..\lib
+xcopy /i /y out32d\*.lib ..\lib
+xcopy /i /y out32d\*.pdb ..\lib
+xcopy /i /y out64\*.lib ..\lib
+xcopy /i /y out64\*.pdb ..\lib
+xcopy /i /y out64d\*.lib ..\lib
+xcopy /i /y out64d\*.pdb ..\lib
+xcopy /e /i /y inc32\*.h ..\include
+:end
diff -ruN openssl-1.0.0b/ms/do_nasm.bat openssl-1.0.0b-patched/ms/do_nasm.bat
--- openssl-1.0.0b/ms/do_nasm.bat	2008-01-06 02:36:21 +0200
+++ openssl-1.0.0b-patched/ms/do_nasm.bat	2010-11-17 23:11:21 +0200
@@ -1,6 +1,6 @@
 
 perl util\mkfiles.pl >MINFO
-perl util\mk1mf.pl nasm VC-WIN32 >ms\nt.mak
+perl util\mk1mf.pl nasm %1 VC-WIN32 >ms\nt.mak
 perl util\mk1mf.pl dll nasm VC-WIN32 >ms\ntdll.mak
 perl util\mk1mf.pl nasm BC-NT >ms\bcb.mak
 
diff -ruN openssl-1.0.0b/ms/do_win64a.bat openssl-1.0.0b-patched/ms/do_win64a.bat
--- openssl-1.0.0b/ms/do_win64a.bat	2010-01-15 17:26:32 +0200
+++ openssl-1.0.0b-patched/ms/do_win64a.bat	2010-11-17 23:18:39 +0200
@@ -2,7 +2,7 @@
 perl util\mkfiles.pl >MINFO
 perl ms\uplink.pl win64a > ms\uptable.asm
 ml64 -c -Foms\uptable.obj ms\uptable.asm
-perl util\mk1mf.pl no-asm VC-WIN64A >ms\nt.mak
+perl util\mk1mf.pl no-asm %1 VC-WIN64A >ms\nt.mak
 perl util\mk1mf.pl dll no-asm VC-WIN64A >ms\ntdll.mak
 
 perl util\mkdef.pl 32 libeay > ms\libeay32.def
diff -ruN openssl-1.0.0b/util/mk1mf.pl openssl-1.0.0b-patched/util/mk1mf.pl
--- openssl-1.0.0b/util/mk1mf.pl	2010-09-02 20:23:36 +0300
+++ openssl-1.0.0b-patched/util/mk1mf.pl	2010-11-17 23:03:18 +0200
@@ -228,8 +228,8 @@
 	$cflags.=' -DTERMIO';
 	}
 
-$out_dir=(defined($VARS{'OUT'}))?$VARS{'OUT'}:$out_def.($debug?".dbg":"");
-$tmp_dir=(defined($VARS{'TMP'}))?$VARS{'TMP'}:$tmp_def.($debug?".dbg":"");
+$out_dir=(defined($VARS{'OUT'}))?$VARS{'OUT'}:$out_def;
+$tmp_dir=(defined($VARS{'TMP'}))?$VARS{'TMP'}:$tmp_def;
 $inc_dir=(defined($VARS{'INC'}))?$VARS{'INC'}:$inc_def;
 
 $bin_dir=$bin_dir.$o unless ((substr($bin_dir,-1,1) eq $o) || ($bin_dir eq ''));
@@ -513,7 +513,7 @@
 EOF
 
 $rules=<<"EOF";
-all: banner \$(TMP_D) \$(BIN_D) \$(TEST_D) \$(LIB_D) \$(INCO_D) headers lib exe
+all: banner \$(TMP_D) \$(BIN_D) \$(TEST_D) \$(LIB_D) \$(INCO_D) headers lib
 
 banner:
 $banner
diff -ruN openssl-1.0.0b/util/pl/VC-32.pl openssl-1.0.0b-patched/util/pl/VC-32.pl
--- openssl-1.0.0b/util/pl/VC-32.pl	2010-09-13 19:28:34 +0300
+++ openssl-1.0.0b-patched/util/pl/VC-32.pl	2010-11-17 23:13:20 +0200
@@ -3,9 +3,6 @@
 # Win64 and WinCE [follow $FLAVOR variable to trace the differences].
 #
 
-$ssl=	"ssleay32";
-$crypto="libeay32";
-
 $o='\\';
 $cp='$(PERL) util/copy.pl';
 $mkdir='$(PERL) util/mkdir-p.pl';
@@ -35,9 +32,10 @@
     $base_cflags= " $mf_cflag";
     my $f = $shlib?' /MD':' /MT';
     $lib_cflag='/Zl' if (!$shlib);	# remove /DEFAULTLIBs from static lib
-    $opt_cflags=$f.' /Ox';
-    $dbg_cflags=$f.'d /Od -DDEBUG -D_DEBUG';
+    $opt_cflags=$f.' /O2 /GL -DNDEBUG';
+    $dbg_cflags=$f.'d /Od -DDEBUG -D_DEBUG /RTC1';
     $lflags="/nologo /subsystem:console /opt:ref";
+    $suffix='64';
 
     *::perlasm_compile_target = sub {
 	my ($target,$source,$bname)=@_;
@@ -116,15 +114,23 @@
     $base_cflags= " $mf_cflag";
     my $f = $shlib?' /MD':' /MT';
     $lib_cflag='/Zl' if (!$shlib);	# remove /DEFAULTLIBs from static lib
-    $opt_cflags=$f.' /Ox /O2 /Ob2';
-    $dbg_cflags=$f.'d /Od -DDEBUG -D_DEBUG';
+    $opt_cflags=$f.' /O2 /GL -DNDEBUG';
+    $dbg_cflags=$f.'d /Od -DDEBUG -D_DEBUG /RTC1';
     $lflags="/nologo /subsystem:console /opt:ref";
+    $suffix='32';
     }
 $mlflags='';
 
-$out_def ="out32";	$out_def.="dll"			if ($shlib);
+if ($debug)
+	{
+	$suffix=$suffix.'d';
+	}
+$ssl="ssleay$suffix";
+$crypto="libeay$suffix";
+
+$out_def ="out$suffix";	$out_def.="dll"			if ($shlib);
 			$out_def.='_$(TARGETCPU)'	if ($FLAVOR =~ /CE/);
-$tmp_def ="tmp32";	$tmp_def.="dll"			if ($shlib);
+$tmp_def ="tmp$suffix";	$tmp_def.="dll"			if ($shlib);
 			$tmp_def.='_$(TARGETCPU)'	if ($FLAVOR =~ /CE/);
 $inc_def="inc32";
 
@@ -138,8 +144,8 @@
 	}
 
 # generate symbols.pdb unconditionally
-$app_cflag.=" /Zi /Fd\$(TMP_D)/app";
-$lib_cflag.=" /Zi /Fd\$(TMP_D)/lib";
+$app_cflag.=" /Zi /Fd$tmp_def/app";
+$lib_cflag.=" /Zi /Fd$out_def/lib$suffix";
 $lflags.=" /debug";
 
 $obj='.obj';
