diff -ruN lzo-2.03/B/win32/vc.bat lzo-2.03-patched/B/win32/vc.bat
--- lzo-2.03/B/win32/vc.bat	Wed Apr 30 15:51:50 2008
+++ lzo-2.03-patched/B/win32/vc.bat	Wed Mar 11 18:18:10 2009
@@ -7,13 +7,58 @@
 @if "%BECHO%"=="n" echo off
 
 
-set CC=cl -nologo -MD
-set CF=-O2 -GF -W3 %CFI% %CFASM%
-set LF=%BLIB%
+set CC=cl 
+set CF=-nologo -Zi -W3 -Gy -GF -MP %CFI% %CFASM%
+set LF=-link -subsystem:console -incremental:no
+@if "%1"=="debug" goto debug
+
+:release
+set CF=%CF% -MT -O2 -GL
+set LF=%LF% -opt:ref -opt:icf -LTCG
+set MASM=ml -nologo -coff -safeseh -c
+set suffix=_32
+
+goto build
+
+:debug
+set CF=%CF% -MTd -Od -RTC1
+set LF=%LF% -debug -fixed:no
+set MASM=ml -nologo -coff -safeseh -Zi -c
+set suffix=_32d
+
+:build
+set BLIB=%BNAME%%suffix%
+set CF=%CF% -Fd%BLIB%.pdb
+set LF=%LF% %BLIB%.lib
+
+%MASM% asm\i386\src_masm\lzo1c_s1.asm
+@if errorlevel 1 goto error
+%MASM% asm\i386\src_masm\lzo1c_s2.asm
+@if errorlevel 1 goto error
+%MASM% asm\i386\src_masm\lzo1f_f1.asm
+@if errorlevel 1 goto error
+%MASM% asm\i386\src_masm\lzo1f_f2.asm
+@if errorlevel 1 goto error
+%MASM% asm\i386\src_masm\lzo1x_f1.asm
+@if errorlevel 1 goto error
+%MASM% asm\i386\src_masm\lzo1x_f2.asm
+@if errorlevel 1 goto error
+%MASM% asm\i386\src_masm\lzo1x_s1.asm
+@if errorlevel 1 goto error
+%MASM% asm\i386\src_masm\lzo1x_s2.asm
+@if errorlevel 1 goto error
+%MASM% asm\i386\src_masm\lzo1y_f1.asm
+@if errorlevel 1 goto error
+%MASM% asm\i386\src_masm\lzo1y_f2.asm
+@if errorlevel 1 goto error
+%MASM% asm\i386\src_masm\lzo1y_s1.asm
+@if errorlevel 1 goto error
+%MASM% asm\i386\src_masm\lzo1y_s2.asm
+@if errorlevel 1 goto error
 
 %CC% %CF% -c @b\src.rsp
 @if errorlevel 1 goto error
-link -lib -nologo -out:%BLIB% @b\win32\vc.rsp
+link -lib -nologo -out:%BLIB%.lib @b\win32\vc.rsp
 @if errorlevel 1 goto error
 
 %CC% %CF% examples\dict.c %LF%
@@ -30,7 +75,7 @@
 %CC% %CF% lzotest\lzotest.c %LF%
 @if errorlevel 1 goto error
 
-%CC% %CF% -Iinclude\lzo minilzo\testmini.c minilzo\minilzo.c
+%CC% %CF% -Iinclude\lzo minilzo\testmini.c minilzo\minilzo.c %LF%
 @if errorlevel 1 goto error
 
 
diff -ruN lzo-2.03/B/win32/vc.rsp lzo-2.03-patched/B/win32/vc.rsp
--- lzo-2.03/B/win32/vc.rsp	Wed Apr 30 15:51:50 2008
+++ lzo-2.03-patched/B/win32/vc.rsp	Wed Mar 11 18:05:08 2009
@@ -65,15 +65,15 @@
 .\lzo_ptr.obj
 .\lzo_str.obj
 .\lzo_util.obj
-.\asm\i386\obj\win32\lzo1c_s1.obj
-.\asm\i386\obj\win32\lzo1c_s2.obj
-.\asm\i386\obj\win32\lzo1f_f1.obj
-.\asm\i386\obj\win32\lzo1f_f2.obj
-.\asm\i386\obj\win32\lzo1x_f1.obj
-.\asm\i386\obj\win32\lzo1x_f2.obj
-.\asm\i386\obj\win32\lzo1x_s1.obj
-.\asm\i386\obj\win32\lzo1x_s2.obj
-.\asm\i386\obj\win32\lzo1y_f1.obj
-.\asm\i386\obj\win32\lzo1y_f2.obj
-.\asm\i386\obj\win32\lzo1y_s1.obj
-.\asm\i386\obj\win32\lzo1y_s2.obj
+.\lzo1c_s1.obj
+.\lzo1c_s2.obj
+.\lzo1f_f1.obj
+.\lzo1f_f2.obj
+.\lzo1x_f1.obj
+.\lzo1x_f2.obj
+.\lzo1x_s1.obj
+.\lzo1x_s2.obj
+.\lzo1y_f1.obj
+.\lzo1y_f2.obj
+.\lzo1y_s1.obj
+.\lzo1y_s2.obj
diff -ruN lzo-2.03/B/win64/vc.bat lzo-2.03-patched/B/win64/vc.bat
--- lzo-2.03/B/win64/vc.bat	Wed Apr 30 15:51:50 2008
+++ lzo-2.03-patched/B/win64/vc.bat	Wed Mar 11 17:50:32 2009
@@ -7,13 +7,31 @@
 @if "%BECHO%"=="n" echo off
 
 
-set CC=cl -nologo -MD
-set CF=-O2 -GF -W3 -Wp64 %CFI%
-set LF=%BLIB%
+set CC=cl 
+set CF=-nologo -Zi -W3 -Gy -GF -MP %CFI%
+set LF=-link -subsystem:console -incremental:no
+@if "%1"=="debug" goto debug
+
+:release
+set CF=%CF% -MT -O2 -GL
+set LF=%LF% -opt:ref -opt:icf -LTCG
+set suffix=_64
+
+goto build
+
+:debug
+set CF=%CF% -MTd -Od -RTC1
+set LF=%LF% -debug -fixed:no
+set suffix=_64d
+
+:build
+set BLIB=%BNAME%%suffix%
+set CF=%CF% -Fd%BLIB%.pdb
+set LF=%LF% %BLIB%.lib
 
 %CC% %CF% -c @b\src.rsp
 @if errorlevel 1 goto error
-link -lib -nologo -out:%BLIB% @b\win64\vc.rsp
+link -lib -nologo -out:%BLIB%.lib @b\win64\vc.rsp
 @if errorlevel 1 goto error
 
 %CC% %CF% examples\dict.c %LF%
@@ -30,7 +48,7 @@
 %CC% %CF% lzotest\lzotest.c %LF%
 @if errorlevel 1 goto error
 
-%CC% %CF% -Iinclude\lzo minilzo\testmini.c minilzo\minilzo.c
+%CC% %CF% -Iinclude\lzo minilzo\testmini.c minilzo\minilzo.c %LF%
 @if errorlevel 1 goto error
 
 
diff -ruN lzo-2.03/build.bat lzo-2.03-patched/build.bat
--- lzo-2.03/build.bat	Thu Jan  1 02:00:00 1970
+++ lzo-2.03-patched/build.bat	Wed Mar 11 18:11:40 2009
@@ -0,0 +1,9 @@
+call "%VCINSTALLDIR%\vcvarsall.bat" x86
+call B\win32\vc.bat debug
+call B\win32\vc.bat
+call "%VCINSTALLDIR%\vcvarsall.bat" x86_amd64
+call B\win64\vc.bat debug
+call B\win64\vc.bat
+xcopy /i /y lzo2_*.lib ..\lib
+xcopy /i /y lzo2_*.pdb ..\lib
+xcopy /e /i /y include\*.h ..\include
